RDS Update RDSH Collection template

mmarch opened the pull request:
New template to update RDSH servers in existing RD Session Host collection to new base image.

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
move to variable

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
rename these to emphasize that they are *existing* resources and must match the VMs

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
rename these to emphasize that they are existing resources and must match the VMs

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
remove blank lines

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
can combine these two into 1:

subnetId : "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnetName'), parameters('subnetName'))]"

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
use resourceId()

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
use json based config instead of XML

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
use literal apiVersions so schema validation can be done

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc for using artifacts in samples

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
not sure what this applies to?   move "rdshVmSize" parameter to variables?   
customer needs to be able select vm size when scheduling template

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
fixed.

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
fixed.

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
fixed this.
are you saying all templates using "apiVersion": "[variables('apiVersion')]",  doing this wrong?  
I've seen so many places where one apiVersion is used across whole template... but I would agree: each resource has it's own version list

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
removed diagnostics extension altogether, it's been causing this much trouble

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
fixed this.

mmarch commented:
hi Brian, I updated template based on review comments,  can you please have another look, and approve

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
No, just use the default size (assuming it's appropriate for the workload - developers that want to change it will know how.  If it *must* match the existing instances then you can remove allowed values since it *must* match (including them creates a maint issue and cause region availability failures)

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
still have a number of blank lines here

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
remove this and use _artifactsLocation - all references to artifacts should use the same pattern.

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
simplify just use resourceId()

resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('existingVnetName')), parameters('existingSubnetName'))

include the resourcegroup param as appropriate

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
does the vnet have to be in the same resourcegroup this template is deployed to?  If so, we should note that in the readme - if not, we should add a param for the RG.

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
is this a prereq?  it's not defined in the template and has a hard coded name...

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
reference this artifact the same was as others (using _artifactsLocation)

bmoore-msft commented:
Check your CI results, I don't think this template can pass deployment (since it relies on existing resources) but it should pass validation...

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
fixed.

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
yes, has to be same RG.

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
ok, thanks for the hint.  fixed.

mmarch commented:
hi Brian.   updated once again.  please review.

the CI was failing because parameter names in azuredeploy.parameters.json were not matching params in the template after I prefixed them with 'existing'.  
I fixed that. Now it's failing because VNET doesn't exist, but I guess that's ok because the assumption is this template is run against existing RG with existing VNET.

Thanks for your help.

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
```diff
@@ -267,7 +266,7 @@
                         "autoUpgradeMinorVersion": true,
                         "settings": {
-                            "ModulesUrl": "[concat(variables('assetLocation'),'/Configuration.zip')]",
+                            "ModulesUrl": "[concat(parameters('_artifactsLocation'),'/Configuration.zip')]",
```
you'll also need a sasToken on here (see the BP doc) since storage is secure by default...

mmarch commented:
added _artifactsLocationSasToken to the URIs.  Brian, please review

bmoore-msft commented:
see the last comment on vmSize param...

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
I'm not sure I understand what you're asking me to do.
'rdshVmSize' already has a default value of 'Standard_A2'
It doesn't have to match existing instances. New vm instances may have a different size than before.

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
Change that parameter to a variable and give it a default size appropriate for the sample.  We want to minimize params that aren't required for a sample to successfully deploy, keeps maint down, and success up.  Dev's who want to change it (or parameterize it) will figure out how.

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
this is the same standard we're applying to all templates, this one doesn't seem to be unique in any sense...  And I have yet to have feedback that this is too challenging for the audience (and have plenty of feedback on the other side).  Happy to chat w/ your PM if it helps.

mmarch commented on rds-update-rdsh-collection/azuredeploy.json:
I have removed the allowedValues list from the rdshVmSize,  
I'm leaving it as a parameter.
I hope this is ok now.

bmoore-msft commented on rds-update-rdsh-collection/azuredeploy.json:
```diff
@@ -33,59 +33,7 @@
-                "Standard_GS4",
-                "Standard_GS5"
-            ]
+            "defaultValue": "Standard_A2"
```
The comment/fix here is to remove the parameter and make it a variable.

mmarch replied:
let me clarify:  you are not allowing me to have a vmSize parameter in my template, and want that as a variable?

mmarch commented:
hi Brian, 
I put the vmSize back in the parameters list as per our conversation, and I think I addressed all the other pr feedback that we had, appreciate if you can review and merge when you get a chance.
Thanks.
